From 171bba4098de705fa2001ceb7fa7fd1fe17860e0 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Fri, 10 Mar 2017 14:54:01 +0800
Subject: [PATCH 2534/5242] MLK-14409-01 ARM: imx: Add speed grading fuse
 check for 900MHz on i.mx6ull

commit  13cbb53f06c461a6baa9ecd34490b7eeda9142ce from
https://source.codeaurora.org/external/imx/linux-imx.git

According to the latest datasheet(Rev.1,02/2017), when the internal LDO
is enabled, the ARM core can run at 900MHz. We need to check the
speed grading fuse to determine the max ARM core frequency.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/mach-imx6ul.c |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-imx/mach-imx6ul.c b/arch/arm/mach-imx/mach-imx6ul.c
index bcdcf7e..851034d 100644
--- a/arch/arm/mach-imx/mach-imx6ul.c
+++ b/arch/arm/mach-imx/mach-imx6ul.c
@@ -72,7 +72,7 @@ static void __init imx6ul_enet_phy_init(void)
 #define OCOTP_CFG3			0x440
 #define OCOTP_CFG3_SPEED_SHIFT		16
 #define OCOTP_CFG3_SPEED_696MHZ		0x2
-#define OCOTP_CFG3_SPEED_1_GHZ		0x3
+#define OCOTP_CFG3_SPEED_900MHZ		0x3
 
 static void __init imx6ul_opp_check_speed_grading(struct device *cpu_dev)
 {
@@ -101,7 +101,7 @@ static void __init imx6ul_opp_check_speed_grading(struct device *cpu_dev)
 	 * 2b'00: Reserved;
 	 * 2b'01: 528000000Hz;
 	 * 2b'10: 700000000Hz(i.MX6UL), 800000000Hz(i.MX6ULL);
-	 * 2b'11: Reserved(i.MX6UL), 1GHz(i.MX6ULL);
+	 * 2b'11: 900000000Hz(i.MX6ULL);
 	 * We need to set the max speed of ARM according to fuse map.
 	 */
 	val = readl_relaxed(base + OCOTP_CFG3);
@@ -115,15 +115,15 @@ static void __init imx6ul_opp_check_speed_grading(struct device *cpu_dev)
 	}
 
 	if (cpu_is_imx6ull()) {
-		if (val != OCOTP_CFG3_SPEED_1_GHZ) {
-			if (dev_pm_opp_disable(cpu_dev, 996000000))
-				pr_warn("Failed to disable 996MHz OPP\n");
-		}
-
 		if (val != OCOTP_CFG3_SPEED_696MHZ) {
 			if (dev_pm_opp_disable(cpu_dev, 792000000))
 				pr_warn("Failed to disable 792MHz OPP\n");
 		}
+
+		if (val != OCOTP_CFG3_SPEED_900MHZ) {
+			if(dev_pm_opp_disable(cpu_dev, 900000000))
+				pr_warn("Failed to disable 900MHz OPP\n");
+		}
 	}
 	iounmap(base);
 
-- 
1.7.9.5

